pkg/util/manifests.go

```diff
@@ -0,0 +1,83 @@
+package util
+
+import (
+	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+// AddEnvVarToDeployment adds an environment variable to the given deployment containers for which the filter function returns true
+func AddEnvVarToDeployment(d *appsv1.Deployment, e v1.EnvVar, appendIfExists bool, filter func(c v1.Container) bool) {
+	ctrs := make([]v1.Container, 0)
+	for _, c := range d.Spec.Template.Spec.Containers {
+		if !filter(c) {
+			ctrs = append(ctrs, c)
+			continue
+		}
+		found := false
+		vars := make([]v1.EnvVar, 0)
+		for _, ce := range c.Env {
+			if ce.Name == e.Name {
+				found = true
+				if appendIfExists {
+					ce.Value = fmt.Sprintf("%s%s", ce.Value, e.Value)
```

Comment:
If the env var exists, I can see how I'd want to control how it's appended.
e.g. Adding a `test` Spring profile should set the value to `test` if it doesn't exist but should append `,test` otherwise.